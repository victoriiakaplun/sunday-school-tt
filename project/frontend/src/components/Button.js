import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function Button() {
  // Bulma class here
  return (
    <div className="button">
      <FontAwesomeIcon css={{ color: 'hotpink', marginRight: '10px' }} icon="crow" />
      :&nbsp; React is Awesome!
    </div>
  );
}

export default Button;
