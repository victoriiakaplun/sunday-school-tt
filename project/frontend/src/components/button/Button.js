import React from 'react';
import classNames from 'classnames';

function Button({ children, onClick, type = 'is-success' }) {
  return (
    <button className={classNames('button', type, 'is-medium')} type="button" onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
