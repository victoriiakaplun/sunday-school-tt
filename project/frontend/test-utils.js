import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

initFontAwesomeLibrary();

// import '@testing-library/jest-dom/extend-expect';

// re-export everything
export * from '@testing-library/react';
