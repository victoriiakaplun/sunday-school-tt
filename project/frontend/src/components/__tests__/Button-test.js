import { render, fireEvent } from 'test-utils';

import Button from '../button/Button';

const props = {
  onClick: jest.fn(),
};

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button {...props}>Click Me!</Button>);

    expect(container.firstChild).toMatchSnapshot(`
      <button
        class="button is-success id-medium"
        type="button" 
      >
        Click Me!
      </button>
    `);
  });

  it('handles onClick event', () => {
    const text = 'Click Me!';
    const onClickMock = jest.fn();
    const { getByText } = render(
      <Button {...props} onClick={onClickMock}>
        {text}
      </Button>,
    );

    fireEvent.click(getByText(text));

    expect(onClickMock).toBeCalledTimes(1);
  });
});
