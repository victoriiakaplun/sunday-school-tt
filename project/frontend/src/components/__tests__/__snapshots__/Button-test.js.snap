// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="button is-success is-medium"
  type="button"
>
  Click Me!
</button>
`;

exports[`Button renders correctly: 
      <button
        class="button is-success id-medium"
        type="button" 
      >
        Click Me!
      </button>
     1`] = `
<button
  class="button is-success is-medium"
  type="button"
>
  Click Me!
</button>
`;
